File: project/DG0/assets/scripts/ddd/Level.lua

require( "scripts/ddd/ilua.lua" );
require( "scripts/ddd/Factory.lua" );
require( "scripts/ddd/Constants.lua" );

--------------------------------------------------------------------------------

Level = { owner = nil, gameID_=0 }

classInheritance( Level, ILua )

--------------------------------------------------------------------------------

function Level:onInit()
	createActor(0, "actor", GT_DEFENCE_GARDEN, LT_MAIN_LEVEL);
	return true;
end

File: project/DG0/src/ddd/src/Level.cpp
#include "ddd/Level.h"

namespace ddd
{
	Level::Level()
		: ILua()
	{
	}

	Level::~Level()
	{
	}

	void Level::onInit()
	{
		removeAllEntity();
		initLuaFunction( 0, "onInit" );
		executeLuaFunction( 0 );
	}

	void Level::onRelease()
	{
		releaseLuaFunction( 0 );
		removeAllEntity();
	}
}